### DISCUSSION TITLE
strongswan neworkmanager gnome charon-nm not able to access crt and key files

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [strongswan/strongswan#2135] [Q&A]
I have no idea where the crt + key needs to be saved so charon-nm can read it. 

`~/.pki`

I read somewhere that this is the location so selinux is not complaining but then still it can't seem to access the files (although it seems to run as root).

**Comment by tobiasbrunner** [comment]
Depends on the distro and its security mechanisms (AppArmor, SELinux etc.). You might have to exclude some files/directories if they aren't already. Read the log for what's happening exactly.

**Comment by f1-outsourcing** [comment]
Yes it is mental, I guess nobody is testing what they are making in this gnome implementation. You can't put them in your home dir and you can't put them (obviously) in a shared dir. 

I even had to enable selinux rules
```
egrep 'example.com.(key|crt)' /var/log/audit/audit.log | audit2allow -M myvpncerts
semodule -i myvpncerts.pp
```

I am also not to sure about this charon/strongswan implementation. I think it tries to access certs as ipsec user. Others like sendmail/apache get security files before they drop privileges.
